pkg/components/loki/lokigrpc/tenant.go

```diff
@@ -0,0 +1,39 @@
+package lokigrpc
+
+import (
+	"context"
+	"errors"
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	lowerOrgIDHeaderName = "x-scope-orgid"
+)
+
+var (
+	ErrDifferentOrgIDPresent = errors.New("different org ID already present")
+	ErrTooManyOrgIDs         = errors.New("multiple org IDs present")
+)
+
+func injectOrgID(ctx context.Context, tenantID string) (context.Context, error) {
+	md, ok := metadata.FromOutgoingContext(ctx)
+	if !ok {
+		md = metadata.New(map[string]string{})
+	}
+	newCtx := ctx
+	if orgIDs, ok := md[lowerOrgIDHeaderName]; ok {
+		if len(orgIDs) == 1 {
+			if orgIDs[0] != tenantID {
+				return ctx, ErrDifferentOrgIDPresent
+			}
+		} else {
+			return ctx, ErrTooManyOrgIDs
+		}
+	} else {
+		md = md.Copy()
+		md[lowerOrgIDHeaderName] = []string{tenantID}
+		newCtx = metadata.NewOutgoingContext(ctx, md)
+	}
```

Comment:
Do we need all the guards here?
If we do, I believe this could be expressed more left-aligned like this?

```suggestion
	orgIDs := md[lowerOrgIDHeaderName]
	switch len(orgIDs) {
	case 0:
		md = md.Copy()
		md[lowerOrgIDHeaderName] = []string{tenantID}
		newCtx = metadata.NewOutgoingContext(ctx, md)
	case 1:
		if orgIDs[0] != tenantID {
			return ctx, ErrDifferentOrgIDPresent
		}
	default:
		return ctx, ErrTooManyOrgIDs
	}
```

Reply:
Honestly, I just copy-pasted from here (for the sake of not adding a new -direct- dependency for just few lines):
https://github.com/weaveworks/common/blob/master/user/grpc.go#L25

But also wondered about its need... 
I'd say currently isn't needed at all, but thought that it could be useful in the future to prevent unexpected situations hard to debug (I guess the purpose of these checks).

However, I'm happy with either applying your suggestions or keeping it simpler for now (I'd probably prefer this for simplicity if that makes sense as well haha).

Thanks! 🙏🏻

Reply:
Choose you, I'm ok with both, I don't see why this could be a problem with the subset we're using 🤔

Reply:
Okay, let's get rid of this until actually needed. Thanks!